qualisys/Qualisys-Unity-SDK
# Custom avatar

[CricketPlayerStreamingTest1.zip](https://github.com/qualisys/Qualisys-Unity-SDK/files/5063288/CricketPlayerStreamingTest1.zip)
[CricketPlayerNB_UnityInit.zip](https://github.com/qualisys/Qualisys-Unity-SDK/files/5063290/CricketPlayerNB_UnityInit.zip)

Support the streaming of custom avatars as well as the QAvatar.  Custom avatars work outside of the Mecanim structure, they take the streaming data without any local rotation modification.

To set up for testing.  The Unity Package has not been updated, so install the old package into your Unity test project then copy the scripts from this branch over the top of those in your project.

To test the new feature:

1. Import the FBX files into Unity.  One is for the player, the other is for the bat.  Both are solved in the QTM file.
2. Add the RTSkeleton scripts to both.  "NB" is the skeleton name for the player, "Bat" is for the bat.  Leave the "Destination Avatar" field blank for both.  Turn OFF the "Use Mecanim" flag.
3. Load the QTM file into QTM and stream it.  
4. Go into Play mode and connect to QTM, the data should be streaming onto the player and bat.

To ensure the legacy QAvatar still works:

- Follow the original instructions for streaming the QAvatar.  The only difference is to make sure the "Use Mecanim" flag is left ON (this is the default so it should already be set to this).
